hack/generated/pkg/reconcilers/azure_deployment_reconciler.go

```diff
@@ -108,20 +108,56 @@ func NewAzureDeploymentReconciler(
 	}
 }
 
+// checkForFatalReconciliationError:
+// - at the moment this is only raised during tests if we determine that a record-replay
+//   test has gone off the rails, but it will be useful for scenarios in the real world as well
+func (r *AzureDeploymentReconciler) checkForFatalReconciliationError(ctx context.Context, err error) error {
```

Comment:
This method is named as though it's a predicate that returns a bool, but does something quite other. 

I'm having trouble coming up with a different name, and I believe that's because the method is doing two different things - checking to see whether there's a fatal error, and storing that error on a condition.

Suggest replacing this method with two. 

Firstly, a conversion method:
``` go
func AsFataReconciliationError(err error) (FatalReconciliationError, bool)
```

Secondly, a method to store the error as a condition:
``` go
func (r *AzureDeploymentReconciler) SetErrorCondition(err error)
```

You can then replace all calls to `checkForFatalReconciliationError` with:
``` go
if fatal, ok := AsFatalReconciliationError(err); ok {
    r.SetErrorCondition(fatal)
}
```

Reply:
right but then I’m going to have to repeat that everywhere… I’ll still split it up into smaller pieces but I want to retain the check & apply method

Reply:
how about `Turn(Appropriate)ErrorsIntoConditions` 😛

ideally we'd have a top-level place where we could do a single check for this and then write it inline like you had, but it's not easy when we are working inside the controller-runtime framework

Reply:
…which I have done!